pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -403,7 +396,15 @@ def barrier(self, *args, **kwargs) -> None:
             torch.distributed.barrier()
 
     def broadcast(self, obj: object, src: int = 0) -> object:
-        return self.dist.broadcast(obj)
+        if not distributed_available():
+            raise RuntimeError(
+                "DDPSpawn is not initialized and torch.distributed is not avalible, can not broadcast object"
+            )
+        obj = [obj]
+        if self.global_rank != 0:
+            obj = [None] * len(obj)
+        broadcast_object_list(obj, src, group=_group.WORLD)
+        return obj[0]
```

Comment:
could we special case tensors here. if the object is tensor type, we call broadcast, otherwise we call broadcast_object_list ?

Reply:
Yeah, should we do the same for all other collectives too? like horovod